lib/zendesk-ticket-deduplicator.rb

```diff
@@ -0,0 +1,97 @@
+require_relative 'zendesk-setup.rb'
+
+require 'rest-client'
+require 'json'
+
+# puts "Total Users on the system: #{@client.users.count}"
+minus_1_hr = Time.now - 2 * 3600
+time_minus_1_hr = minus_1_hr.strftime('%Y-%m-%dT%H:%M:%S%z')
+
+puts "time: #{time_minus_1_hr}"
+
+# Change date to 2020-01-01
+search_results = @client.search(:query => "type:user role:end-user -name:Zendesk organization:none created_at>=#{time_minus_1_hr}")
+
+user_count =  search_results.count
+number_of_pages = (user_count.to_f / 100).ceil
+
+puts "Number of User accounts (UIDs): #{user_count}"
+puts "Number of pages of UIDs: #{number_of_pages}"
+puts "Retrieving #{user_count} user accounts, this may take a while"
+
+# Reset name after testing - Set Logfile name
+results_file = "data/2020_user_ids.json"
+
+# Loop through users and write to file in JSON format
+File.open(results_file, "w") do |file|
+  (1..number_of_pages).each do |i|
+    search_results.page(i).each do |user|
+      file.puts(user.to_json)
```

Comment:
One change you might want to consider here: if you make this file a valid json document, you can simplify the "read it in" logic below to make more use of the json library, instead of reading line-by-line as you are at present.

Reply:
I"ll think about this for the next update. Thanks @blairboy362